ci/docker/linux-apt-lint.dockerfile

```diff
@@ -56,20 +56,8 @@ COPY ci/etc/rprofile /arrow/ci/etc/
 RUN cat /arrow/ci/etc/rprofile >> $(R RHOME)/etc/Rprofile.site
 # Also ensure parallel compilation of C/C++ code
 RUN echo "MAKEFLAGS=-j$(R -s -e 'cat(parallel::detectCores())')" >> $(R RHOME)/etc/Renviron.site
-
-
-COPY ci/scripts/r_deps.sh /arrow/ci/scripts/
-COPY r/DESCRIPTION /arrow/r/
-# We need to install Arrow's dependencies in order for lintr's namespace searching to work.
-# This could be removed if lintr no longer loads the dependency namespaces (see issues/PRs below)
```

Comment:
@jonkeane which of these issues was about dependency namespaces? I didn't look closely but none were obvious from the title.

I removed arrow dependencies and tested locally, and lintr 3.0.0 seems to work fine, assuming it wasn't silently passing.

Reply:
The namespace issue wasn't one of the ones I raised — it was just a consequence of the way lintr was looking up signatures for functions when linting. Though that code was under a decent amount of flux, so I wouldn't be surprised if it no longer requires the dependencies to be installed. 

If it passes without installing them, we're totally fine to remove it IMO.